lxd/api_cluster.go

```diff
@@ -164,12 +225,69 @@ func clusterPutJoin(d *Daemon, req api.ClusterPut) Response {
 	// Asynchronously join the cluster.
 	run := func(op *operation) error {
 		logger.Debug("Running cluster join operation")
-		// First request for this node to be added to the list of
-		// cluster nodes.
+
+		// If the user has provided a cluster password, setup the trust
+		// relationship by adding our own certificate to the cluster.
+		if req.ClusterPassword != "" {
+			err = cluster.SetupTrust(string(cert.PublicKey()), req.ClusterAddress,
+				string(req.ClusterCertificate), req.ClusterPassword)
+			if err != nil {
+				return errors.Wrap(err, "failed to setup cluster trust")
```

Comment:
Errors must start with a capital (and never end with a period)

Reply:
Fixed.